sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/typedaggregators.scala

```diff
@@ -76,26 +76,126 @@ class TypedCount[IN](val f: IN => Any) extends Aggregator[IN, Long, Long] {
 
   // Java api support
   def this(f: MapFunction[IN, Object]) = this((x: IN) => f.call(x).asInstanceOf[Any])
+  
   def toColumnJava: TypedColumn[IN, java.lang.Long] = {
     toColumn.asInstanceOf[TypedColumn[IN, java.lang.Long]]
   }
 }
 
+class TypedAverage[IN](val f: IN => Double)
+  extends Aggregator[IN, (Double, Long), Double] {
 
-class TypedAverage[IN](val f: IN => Double) extends Aggregator[IN, (Double, Long), Double] {
   override def zero: (Double, Long) = (0.0, 0L)
   override def reduce(b: (Double, Long), a: IN): (Double, Long) = (f(a) + b._1, 1 + b._2)
-  override def finish(reduction: (Double, Long)): Double = reduction._1 / reduction._2
```

Comment:
ditto

Reply:
Order of functions is consistent among all aggregation functions: zero, reduce, merge finish.  Hence the swap of location of the merge and finish functions